compiler/src/dotty/tools/dotc/transform/PatternMatcher.scala

```diff
@@ -719,16 +719,19 @@ object PatternMatcher {
           val expectedTp = tpt.tpe
 
           // An outer test is needed in a situation like  `case x: y.Inner => ...`
-          def outerTestNeeded: Boolean =
-            // See the test for SI-7214 for motivation for dealias. Later `treeCondStrategy#outerTest`
-            // generates an outer test based on `patType.prefix` with automatically dealises.
-            expectedTp.dealias match {
+          def outerTestNeeded: Boolean = {
+            def go(expected: Type): Boolean = expected match {
```

Comment:
Might neeed to dealias here too if the constructor is an alias.

Reply:
I think I added a test for that now.